### ISSUE TITLE
Incorrect behavior for CSS files with async imports

### ISSUE THREAD

**Issue by Ajaxy** [parcel-bundler/parcel#4497]
# 🐛 bug report

The bug happens when the following conditions are met:

1. There is a `module.js` which depends on `module.css` and is statically imported into both `index.js` and `async.js`.
2. There is an `importAsync.js` which provides a function that is called in `index.js`. This function dynamically imports `async.js`.
3. `importAsync.js` is imported _before_ `module.js` in `index.js`.

[`index.js`](https://github.com/Ajaxy/parcel-bug-async-css/blob/master/src/index.js):
```js
import importAsync from './importAsync';

// Inserted after `importAsync`
import './module';

import './index.css';

setTimeout(() => {
  // Works fine. Only one <link> added into `index.html`
  // import('./async');

  // Additional `async.*.css` added into `index.html`
  importAsync();
}, 1000);
```

[`index.js`](https://github.com/Ajaxy/parcel-bug-async-css/blob/master/src/importAsync.js):
```js
export default function () {
  import('./async');
}
```

## 🤔 Expected Behavior

`module.css` is merged into a single `app.css` bundle.

## 😯 Current Behavior

Two CSS files are created, and both added into [`index.html`](https://github.com/Ajaxy/parcel-bug-async-css/blob/master/build/index.html#L1).

## 🔦 Context

I need more control when importing modules, that's why I'm using a proxy function.

## 💻 Code Sample

https://github.com/Ajaxy/parcel-bug-async-css

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.226
| Node             | v12.13.0
| npm/Yarn         | 6.12.0
| Operating System | macOS 10.14.16

**Comment by mischnic** [comment]
So the expected behaviour would be:
- Create a CSS for module.css and index.css (this should be achievable by modify the DefaultBundler).
- Should `import`ing async.js load the module/index CSS bundle again? The order for css imports from JS isn't really specified anywhere.

**Comment by Ajaxy** [comment]
> Should importing async.js load the module/index CSS bundle again?

I don’t think so, since both are eventually imported into `index.js`, so the `module.css` will be ready by the moment the `async` is imported.

**Comment by mischnic** [comment]
One could also make the argument that loading a css asset after another css asset (again) should overwrite the previous one...
So importing the CSS assets X, Y and then an async import that uses the CSS assets X, Z should cause X to overwrite Y (meaning we have to load it again)

**Comment by Ajaxy** [comment]
Well, in my case I have a `Button` component, which has its styles in `Button.css`. Then I'm using this component across the app, including the async components. Since the `Button.js` code is included in the main JS bundle, I expect the `Buttons.css` to be also included in the main CSS bundle. I don't want it to be loaded again as I'm trying to reduce the bundle size.

What is the case of one's need to overwrite the previous CSS?

Again, everything works just fine if in the `index` I do `import module` _before_ anything related to `async`. I'm assuming that the parser first sees the `module.css` related to the main bundle and does not include it into `async` one (which is correct in my case). So it goes `index.js -> module.js -> module.css -> importAsync.js -> async.js`.

If I change the order of imports (as in my first example), it would go `index.js -> importAsync.js -> async.js -> module.js -> module.css` meaning it first finds `module.css` through `async.js`.
